kube-client/src/client/builder.rs

```diff
@@ -104,14 +107,36 @@ impl TryFrom<Config> for ClientBuilder<GenericService> {
             }
 
             #[cfg(feature = "http-proxy")]
-            Some(proxy_url) if proxy_url.scheme_str() == Some("http") => {
+            Some(proxy_url) if proxy_url.scheme_str() == Some(PROXY_SCHEME_HTTP) => {
                 let proxy = hyper_http_proxy::Proxy::new(hyper_http_proxy::Intercept::All, proxy_url.clone());
                 let connector = hyper_http_proxy::ProxyConnector::from_proxy_unsecured(connector, proxy);
 
                 make_generic_builder(connector, config)
             }
 
-            _ => make_generic_builder(connector, config),
+            proxy_url => {
+                if let Some(proxy_url) = proxy_url {
+                    let missing_proxy_feature = match proxy_url.scheme_str() {
+                        Some(PROXY_SCHEME_SOCKS5) => Some("kube/socks5"),
+                        Some(PROXY_SCHEME_HTTP) => Some("kube/http-proxy"),
+                        _ => None,
+                    };
+                    if let Some(missing_proxy_feature) = missing_proxy_feature {
+                        tracing::warn!(
```

Comment:
Should these be hard errors instead? "We know what you're trying to do but it's not supported so let's do the wrong thing" isn't great...

Reply:
this feels like a place where a `panic!` is justified to me. we want this to be noticed pretty quickly, and it's a local development convenience anyway.

Reply:
Converted to errors in e81e3ac8f17d4e1e8516b54e401e91f60e8cdc28